tools/cldr-code/src/main/java/org/unicode/cldr/util/GrammarInfo.java

```diff
@@ -36,16 +36,17 @@
     public enum GrammaticalTarget {nominal}
 
     /**
-     * The ordering of these values is intended to put the default values first, and to group values together that tend to have similar forms.
+     * The ordering of these values is intended to put the default values first, and to group values together that tend to have similar forms for the most common cases,
+     * then have the rest in alphabetical order.
      */
-    public enum CaseValues {nominative, vocative, accusative, oblique, genitive, dative, locative, instrumental, prepositional,
-        ablative, adessive, allative, causal, delative, elative, essive, illative, inessive, sublative, superessive, terminative, translative;
+    public enum CaseValues {nominative, vocative, accusative, oblique, genitive, dative, locative, instrumental, prepositional, ablative,
+        abessive, adessive, allative, causal, comitative, delative, elative, ergative, essive, illative, inessive, locativecopulative, partitive, sociative, sublative, superessive, terminative, translative;
```

Comment:
is this verified against DTD / supplemental metadata?

Reply:
Yes, there is a general verification of the valid values in the DTD. This is to align the enum with the DTD values.

I've been thinking for some time that we could simplify our lives if we had a simple program that generated enums from the valid DTD values (both @match values and the regular DTD options). Don't know whether it would be worth the extra step of running it whenever we change the DTD.